lib/oidcstrategy.js

```diff
@@ -172,6 +219,44 @@ util.inherits(Strategy, passport.Strategy);
  * @api protected
  */
 Strategy.prototype.authenticate = function authenticateStrategy(req, options) {
+  /* 
+   * We should be careful using 'this'. Avoid the usage like `this.xxx = ...`
+   * unless you know what you are doing.
```

Comment:
Is this a generic comment on JavaScript, or something specific to the OIDC code.  If it is the former I would remove it.